api/webhooks/views.py

```diff
@@ -1,302 +1,180 @@
-
-            # Order is already in a state where no updates are needed
-            if payment_order.status not in needs_update_statuses:
-                return Response(status=200)
-
-            # Check payment status from the API
-            payment = get_payment(remote_id, settings.VERKKOKAUPPA_NAMESPACE)
-            payment_status = getattr(payment, "status", "payment_not_found")
-
-            if payment_status != "payment_paid_online":
-                capture_message(
-                    f"Received payment webhook for order {remote_id} that is not paid: {payment_status}",
-                    level="warning",
-                )
-                raise WebhookError(message="Invalid payment state", status_code=400)
-
-            if payment_order.status in needs_update_statuses:
-                payment_order.status = OrderStatus.PAID
-                payment_order.payment_id = payment_id
-                payment_order.processed_at = datetime.now().astimezone(get_default_timezone())
-                payment_order.save()
-
-            reservation: Reservation = payment_order.reservation
-            if reservation.state == ReservationStateChoice.WAITING_FOR_PAYMENT:
-                reservation.state = ReservationStateChoice.CONFIRMED
-                reservation.save()
-                send_confirmation_email(reservation)
-
-            return Response(status=200)
-        except WebhookError as e:
-            with push_scope() as scope:
-                scope.set_extra("details", "Invalid payment webhook")
-                scope.set_extra("data", e.to_json())
-                scope.set_extra("status_code", e.status_code)
-                capture_exception(e)
-            return Response(data=e.to_json(), status=e.status_code)
-        except GetPaymentError as e:
-            with push_scope() as scope:
-                scope.set_extra("details", "Checking order payment failed")
-                scope.set_extra("remote_id", remote_id)
-                capture_exception(e)
-            return Response(
-                data={"status": 500, "message": "Problem with upstream service"},
-                status=500,
-            )
-
-
-class WebhookOrderViewSet(viewsets.ViewSet):
+            payment = get_payment(order_id, namespace)
+        except GetPaymentError:
+            msg = f"Checking payment for order '{order_id}' failed"
+            capture_message(msg)
+            return Response(data={"message": msg}, status=500)
+
+        if payment is None:
+            msg = f"Payment '{order_id}' not found from verkkokauppa"
+            capture_message(msg)
+            return Response(data={"message": msg}, status=404)
+
+        if payment.status != PaymentStatus.PAID_ONLINE.value:
+            msg = f"Invalid payment status: '{payment.status}'"
+            capture_message(msg)
+            return Response(data={"message": msg}, status=400)
+
+        # TODO: Should this be enforced?
```

Comment:
No need to enforce, it is desirable to still track the status of the payment even if the reservation which spawned it has been lost (has expired because not paid with time limits).

Reply:
Ok, removed comment